sirepo/template/template_common.py

```diff
@@ -83,6 +85,48 @@ def __init__(self, **kwargs):
         )
 
 
+class LogParser(PKDict):
+    def __init__(self, run_dir, log_filename, **kwargs):
+        super().__init__(run_dir=run_dir, log_filename=log_filename, **kwargs)
+        self.pksetdefault(
+            error_patterns=_DEFAULT_ERROR_PATTERNS,
+            multiline_error_pattern=r"^Traceback .*?^\w*Error: (.*?)\n",
+            default_msg="An unknown error occurred",
+        )
+
+    def parse_log_file_for_errors(self):
+        res = ""
+        p = self.run_dir.join(self.log_filename)
+        if not p.exists():
+            return res
+        with pkio.open_text(p) as f:
+            res = self._parse_log(f, res)
+        if res:
+            return res
+        return self.default_msg
+
+    def parse_multiline(self):
```

Comment:
I'm not 100% positive (so please look into the code) but I don't think this needs to be a special case. I think for mpi we can define a `class _MPILogParser` and override `_parse_log` with this. We can set `default_msg` to be `None`. Then I think the only case that would be a change in behavior with the existing implementation is what we return in the log file not existing case. Currently we return `None`. If we use parse_log_file_for_errors() we will return `""`. But, looking at the js code we decide to display an alert based on the truthiness of the message. Both `None` and `""` are falsey so in the end there won't be an actual change in behavior.